### ISSUE TITLE
DirectLine domain one endpoint websocket inside - standard endpoint

### ISSUE THREAD

**Issue by vglu** [microsoft/BotFramework-WebChat#3188]
Hi,

Example from https://microsoft.github.io/BotFramework-WebChat/01.getting-started/g.hybrid-react-npm/

Code from g.hybrid-react-npm\packages\chat-component\src\ChatComponent.js

DirectLine call with parameters

    useMemo(async () => {
      const domain = 'myDomain/v3/directline';
      const res = await fetch(`${domain}/tokens/generate`, { 
          method: 'POST',
          headers: {
            Authorization: 'Bearer AAA',
          }
     });
  
      const { token } = await res.json();
  
      setDirectLine(createDirectLine({ token, domain }));
    }, []);

method `generate` works fine:

```json
Request URL: myDomain/v3/directline/tokens/generate
Request Method: POST
Status Code: 200 OK
Remote Address: myDomain
Referrer Policy: no-referrer-when-downgrade
```

method `conversations` works fine:

```json
Request URL: myDomain/v3/directline/conversations
Request Method: POST
Status Code: 201 Created
Remote Address: myDomain
Referrer Policy: no-referrer-when-downgrade
```

websocket works different way: 

```json
Request URL: wss://directline.botframework.com/v3/directline/conversations/HfNJdMeadL6DBcmWQlWAbw-h/stream?watermark=1
Request Method: GET
Status Code: 101 Switching Protocols
```
If I turn off websocket (webSocket: true) - all works fine but I prefer use socket.

My clients has no access to internet, and my solution will connect to `nginx` `myDomain` proxy.

Regards,
Vitaliy

**Comment by compulim** [comment]
The Web Socket URL is a value from the /v3/directline/conversation call:

```json
{
  "conversationId": "K4X7vWGMB...",
  "token": "ew0KICAiYWxnIjogIlJ...",
  "expires_in": 3600,
  "streamUrl": "wss://directline.botframework.com/v3/directline/conversations/K4X7vWGMB.../stream?watermark=-&t=ew0KICAiYWxnIjogIlJ...",
  "referenceGrammarId": "12345678-1234-5678-abcd-12345678abcd"
}
```

For client without Internet access, you can also consider using [Direct Line App Service Extensions](https://docs.microsoft.com/en-us/azure/bot-service/bot-service-channel-directline-extension?view=azure-bot-service-4.0), which is targeting this type of users.

**Comment by vglu** [comment]
Hi @compulim 

Direct Line App service Extension. 
First of all - thank you for your advise - I really appreciate it.
Let me detail explain my plan.
I have got standard 

1 QnA service
2 Published bot from QnA service (without any changes)
3 One intranet page call DirectLine with  `domain = 'myDomain/v3/directline`
4 One nginx proxy server with rules to redirect myDomain to Azure bot service

Just only one my local page with changes, all others from the box. This mean, upgrade for new version, bug fixing, support all this services I have got from MS.

Now, if I'll use `Direct Line App service Extension` 

Yours suggestion
If I'll use it - I should place on-prem one more server. In case with proxy - I have got only proxy server and all others (support, traffic, and so on) I have got without changes from Azure. In my case it is - QnA + publish bot (without any changes) and one page for users. In your suggestion I should modify bot and more more .... 

Additional not too geed describe how to setup this functionality from begin to end. May be you have additional resources where I can read and try your solution? 

Lets return back to question .... 

If I pass my domain to DirectLine class - why I have got connection to external socket in response? 

Thanks

Regards,
Vitaliy
